master/buildbot/process/build.py

```diff
@@ -395,99 +400,103 @@ def startBuild(self, workerforbuilder: AbstractWorkerForBuilder):
-            )
-            yield self.master.data.updates.add_build_locks_duration(
-                self.buildid, duration_s=locks_acquired_at - locks_acquire_start_at
-            )
             yield self.master.data.updates.setStepStateString(
-                self._locks_acquire_step.stepid, "locks acquired"
-            )
-            yield self.master.data.updates.finishStep(
-                self._locks_acquire_step.stepid, SUCCESS, False
+                self._preparation_step.stepid, f"worker {self.getWorkerName()} ready"
             )
+            yield self.master.data.updates.finishStep(self._preparation_step.stepid, SUCCESS, False)
+
+            self.conn = workerforbuilder.worker.conn
+
+            # To retrieve the worker properties, the worker must be attached as we depend on its
+            # path_module for at least the builddir property. Latent workers become attached only after
+            # preparing them, so we can't setup the builddir property earlier like the rest of
+            # properties
+            self.setupWorkerProperties(workerforbuilder)
+            self.setupWorkerForBuilder(workerforbuilder)
+            self.subs = self.conn.notifyOnDisconnect(self.lostRemote)
+
+            # tell the remote that it's starting a build, too
+            try:
+                yield self.conn.remoteStartBuild(self.builder.name)
+            except Exception:
+                yield self.buildPreparationFailure(Failure(), "start_build")
+                yield self.buildFinished(["worker", "not", "building"], RETRY)
+                return
+
+            if self._locks_to_acquire:
+                yield self.master.data.updates.setBuildStateString(self.buildid, "acquiring locks")
+                locks_acquire_start_at = int(self.master.reactor.seconds())
+                yield self.master.data.updates.startStep(
+                    self._locks_acquire_step.stepid, started_at=locks_acquire_start_at
+                )
+                yield self.acquireLocks()
+                locks_acquired_at = int(self.master.reactor.seconds())
+                yield self.master.data.updates.set_step_locks_acquired_at(
+                    self._locks_acquire_step.stepid, locks_acquired_at=locks_acquired_at
+                )
+                yield self.master.data.updates.add_build_locks_duration(
+                    self.buildid, duration_s=locks_acquired_at - locks_acquire_start_at
+                )
+                yield self.master.data.updates.setStepStateString(
+                    self._locks_acquire_step.stepid, "locks acquired"
+                )
+                yield self.master.data.updates.finishStep(
+                    self._locks_acquire_step.stepid, SUCCESS, False
+                )
 
         yield self.master.data.updates.setBuildStateString(self.buildid, 'building')
 
+        if self.skipBuild:
+            self.results = SKIPPED
+
         # start the sequence of steps
         self.startNextStep()
```

Comment:
I think we shouldn't attempt to execute steps. Since `.conn` is not set, `startNextStep()` will execute `allStepsDone` immediately. Do we even want steps to be created at all?

Reply:
Probably we shouldn't create steps. This also should be lighter on the database.

Reply:
I was hoping we could get them to show up the UI just for consistency, but yes not much point beyond "it shows up"